arches/app/search/search_export.py

```diff
@@ -93,11 +98,15 @@ def return_ordered_header(self, graphid, export_type):
 
         subcard_list_with_sort.sort(key=lambda x: x.sortorder, reverse=True)
 
-        self.insert_subcard_below_parent_card(sorted_card_list, card_list_no_sort)
+        # Loop down through the levels of subcards, 'till no more sub-levels may be added.
+        subcards_added = [True]
+        while subcards_added[0]:
+            subcards_added = [False]
+            self.insert_subcard_below_parent_card(sorted_card_list, card_list_no_sort, subcards_added)
 
-        # Cards in subcard_list_with_sort are added after cards with no sort
+            # Cards in subcard_list_with_sort are added after cards with no sort
 
-        self.insert_subcard_below_parent_card(sorted_card_list, subcard_list_with_sort)
+            self.insert_subcard_below_parent_card(sorted_card_list, subcard_list_with_sort, subcards_added)
```

Comment:
@khodgkinson-he This works like a charm, but just be consistent with other parts of Arches, could you change this to use recursion rather than a while loop? Something this should work: 
```
def order_cards(subcards_added=True):
        if subcards_added == True:
            subcards_added = False
            x = self.insert_subcard_below_parent_card(sorted_card_list, card_list_no_sort, subcards_added)
            y = self.insert_subcard_below_parent_card(sorted_card_list, subcard_list_with_sort, subcards_added)
            order_cards(y)

    order_cards()
```

With that change you would need to change line 71 to `subcards_added = True` and then add a return statement to line 74, `return subcards_added`

Reply:
@chiatt Glad you like.. ^_^

I've refactored that as an inline function as recommended, any issues please let me know!